<?xml version="1.0" encoding="UTF-8" standalone="yes"?>

<testsuite name="Verify Object in Context">
	<test name="Simple Store Default Object Test" testCaseID="JTAF-420 TC5001" >
		<desc>Test that a calling a function produces the expected result</desc>
		<coverage>JTAF-420</coverage>
		<automationValue>High</automationValue>
		<teststeps>
	    <storeDefaultObject contextKey="verifyKey" default="Test value in context" />
	    <verifyObjectInContext actualKey="verifyKey" expectedKey="testKey" testKey="Test value in context" />
	    <storeDefaultObject contextKey="verifyKey" default="Try new default" />
	    <verifyObjectInContext actualKey="verifyKey" expectedKey="testKey" testKey="Test value in context" />
		</teststeps>
	</test>
	
	<test name="Store Default List Object Test" testCaseID="JTAF-420 TC5002" >
		<desc>Test that a calling a function produces the expected result</desc>
		<coverage>JTAF-420</coverage>
		<automationValue>High</automationValue>
		<teststeps>
	    <storeDefaultObject contextKey="verifyKey">
	      <list name="default">
	        <string>Test value in context</string>
	      </list>
	    </storeDefaultObject>
	    <verifyObjectInContext actualKey="verifyKey" expectedKey="testKey">
	      <list name="testKey">
	        <string>Test value in context</string>
	      </list>
	    </verifyObjectInContext>
	    <storeDefaultObject contextKey="verifyKey">
	      <list name="default">
	        <string>Try new default</string>
	      </list>
	    </storeDefaultObject>
	    <verifyObjectInContext actualKey="verifyKey" expectedKey="testKey">
	      <list name="testKey">
	        <string>Test value in context</string>
	      </list>
	    </verifyObjectInContext>
		</teststeps>
	</test>
	
	
	
					
					
	<test name="verifyObjectInContext - Exception" >
		<automationValue>High</automationValue>
		<teststeps>
			<verifyException>
				<string name = "exceptionType">AssertionError</string>
				<storeDefaultObject contextKey="verifyKey" default="Test" />
				<verifyObjectInContext actualKey="verifyKey" expectedKey="testKey" testKey="Test value in context" title="Testing"/>
			</verifyException>
		</teststeps>
	</test>
	
	<test name="verifyObjectInContext - No match" >
		<automationValue>High</automationValue>
		<teststeps>
			<verifyException>
				<string name = "exceptionType">AssertionError</string>
				<storeDefaultObject contextKey="verifyKey" default="Test" />
				<verifyObjectInContext actualKey="verifyKey" expectedKey="testKey" testKey="Test value in context"/>
			</verifyException>
		</teststeps>
	</test>
	
</testsuite>
